---
title: "Changes in structure and assembly of a species-rich soil natural community with contrasting nutrient availability upon establishment of a plant-beneficial Pseudomonas in the wheat rhizosphere"
author: "Daniel Garrido-Sanz (daniel.garridosanz@unil.ch)"
affiliation: "Kell Lab | Department of fundamental microbiology | University of Lausanne"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  rmdformats::html_clean:
    code_folding: hide
    toc: true
    thumbnails: false
    toc_depth: 3
pkgdown:
  as_is: true
---

This is a modification of the code and data analyses from: https://github.com/dgarrs/Pprotegens_proliferation_NatComs
This is intended for being executed on a (Renku)[https://renkulab.io/] session.

Only the following data analyses are included here:
  1. 

For more details, see the original code and publication:
Garrido-Sanz, D., Čaušević, S., Vacheron, J. et al. Changes in structure and assembly of a species-rich soil natural community with contrasting nutrient availability upon establishment of a plant-beneficial Pseudomonas in the wheat rhizosphere. Microbiome 11, 214 (2023). https://doi.org/10.1186/s40168-023-01660-5



# 1. Load required packages

```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(data.table)
library(ggtree)
library(phyloseq)
library(Biostrings)
library(pals)
library(ggpubr)
library(scales)
library(picante)
library(ggh4x)
library(agricolae)
```

# 2. Load DADA2 results, phylogeny, and metadata

```{r}
taxa2 = read.table(file="Taxtable_dada2.txt", sep=";", header=T)
taxa2 = column_to_rownames(as.data.frame(taxa2), "X")
#seqtab.nochim2 = read.table(file="ASV_sequences.txt", sep=";", header=T)
seqtab.nochim2 = fread(file="ASV_sequences.txt", sep=";", header=T) #fastes
seqtab.nochim2 = column_to_rownames(as.data.frame(seqtab.nochim2), "V1") #V1 if fread used, X if read.table
rownames(seqtab.nochim2) = gsub("_$", "", rownames(seqtab.nochim2))
```

```{r}
tree = read.tree("seqtab.nochim_aln.fasta.contree")
#Rename tip labels
index = data.frame(seqs = rownames(taxa2))
index = rownames_to_column(index, "index")
tip_names = data.frame(index = tree$tip.label)
tip_tax = left_join(tip_names, index, by = "index")
tree$tip.label = tip_tax$seqs
```


```{r}
metadata = read.table("metadata.txt", header = T, sep = "\t")
rownames(metadata) = metadata$id_samples
head(metadata)
```

# 3. Create the phyloseq object

```{r, warnings = FALSE, error = FALSE, message = FALSE}
ps <- phyloseq(otu_table(seqtab.nochim2, taxa_are_rows=F), 
               sample_data(metadata), 
               tax_table(as.matrix(taxa2)),
               phy_tree(tree))

#Store the ASVs DNA sequences the refseq slot of the phyloseq object, and then rename our taxa to a short string.
dna <- Biostrings::DNAStringSet(taxa_names(ps))
names(dna) <- taxa_names(ps)
ps <- merge_phyloseq(ps, dna)
taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))

ps

#write the refseqs
#write.table(refseq(ps), "ref-seq_ps.txt", sep = "\t", quote = F, col.names = F)
```

# 4. Identify *P. protegens* CHA0 ASV

We will use the control; only inoculated with CHA0

```{r, message=FALSE, warning=FALSE}
cntCHA0=subset_samples(ps, ComState=="Control")
cntCHA0=filter_taxa(cntCHA0, function(x) sum(x) > 0, TRUE)
cntCHA0= subset_taxa(cntCHA0, Genus=="Pseudomonas")
otu_table(cntCHA0)
tax_table(cntCHA0)
print(paste0("Pseudomonas protegens CHA0 ASV is: ", colnames(otu_table(cntCHA0)[1,1])))
```
Therefore CHA0 is **ASV1148!**

# 5. Phyloseq filtering and normalization

## 5.1 Remove Mitochondria and Chloroplasts and the control used for identify CHA0 ASV.

```{r, warnings = FALSE, error = FALSE}
#Filter out mitochondria and chloroplasts
ps = subset_taxa(ps, Genus!="Mitochondria")
ps = subset_taxa(ps, Genus!="Chloroplast")
#And the control!
ps = subset_samples(ps, ComState !="Control")
ps
```

## 5.2 Prevalence filtering

Adapted from: [Bioconductor Workflow for Microbiome Data Analysis: from raw reads to community analyses](https://f1000research.com/articles/5-1492/v2)

```{r }
prev0 = apply(X = otu_table(ps),
                MARGIN = ifelse(taxa_are_rows(ps), yes = 1, no = 2),
                FUN = function(x){sum(x > 0)})

prevdf = data.frame(Prevalence = prev0,
                      TotalAbundance = taxa_sums(ps),
                      tax_table(ps))

#only for reference, in case we want to filter Phyla that appear less than X times
keepPhyla = table(prevdf$Phylum)[(table(prevdf$Phylum) > 0)]

prevdf1 = subset(prevdf, Phylum %in% names(keepPhyla))

# Define prevalence threshold as 0.5% of total samples
prevalenceThreshold = 0.005 * nsamples(ps)
prevalenceThreshold

# Execute prevalence filter, using `prune_taxa()` function
ps1 = prune_taxa((prev0 > prevalenceThreshold), ps)
ps1

# Filter entries with unidentified Phylum.
ps2 = subset_taxa(ps1, Phylum %in% names(keepPhyla))
ps2

ggplot(prevdf1, aes(TotalAbundance, Prevalence, color = Phylum)) +
  geom_hline(yintercept = prevalenceThreshold, alpha = 0.5, linetype = 2) +
  geom_point(size = 2, alpha = 0.65) +
  scale_y_log10() + scale_x_log10() +
  xlab("Total Abundance") +
  scale_color_manual(values=ocean.balance(14))+
  theme_bw()+
  facet_wrap(~Phylum, nrow = 4)

```


```{r}
print(paste0("Initial #ASVs:", ntaxa(ps), ". #ASVs after prevalence filtering:", ntaxa(ps2), ". #ASVs removed: ", ntaxa(ps)-ntaxa(ps2)))
```

Convert to relative abundance and store for further inspection

```{r}
ps2.prop = transform_sample_counts(ps2, function(x) {x/sum(x)*100})
#write.table(otu_table(ps2.prop), "otu_table_ps2_prop.txt", quote = F, sep = "\t")
```


## 5.3 Normalization

For certain downstream analyses; for example to compare different abundances of taxa between samples, the counts needs to be normalized to account for different sampling efforts, exposures, baselines, etc. We will normalize the samples using the CSS (Cumulative Sum Scaling) introduced in [Paulson et al. 2013](https://doi.org/10.1038/nmeth.2658), where the offset of a sample is the cumulative sum of counts in that sample, up to a quantile determined in a data driven way. Calculates scaling factors as the cumulative sum of ASVs abundances up to a data-derived threshold.

```{r fig.width = 11, fig.height= 11,}
#Ussing the CSS
ps2.css = metagMisc::phyloseq_transform_css(ps2, norm = T, log = F)
ps2.css = orient_taxa(ps2.css, "columns") #change taxa to columns
#and the the TSS (proportion; relative abundance)
ps2.css.prop = transform_sample_counts(ps2.css, function(x) {x/sum(x)*100})
```

# 6. Barplots

Using normalized relative abundances

```{r}
#Represent only the top 100 ASVs
toptax = names(sort(taxa_sums(ps2.css), decreasing=TRUE))[1:100]
ps.toptax = transform_sample_counts(ps2.css, function(x) {x/sum(x)*100})
ps.toptax <- prune_taxa(toptax, ps.toptax)

#Plot at the class level
plot_bar(prune_taxa(toptax, ps2.css.prop), x="id_samples", fill="Class")+
  geom_bar(aes(fill=Class), stat="identity", position="stack")+
  scale_fill_manual(values=ocean.balance(11))+
  xlab("") + ylab("Normalized relative abundance (%)") +
  facet_wrap(~ComState+Environment+Inoculant, scales="free_x", ncol = 4) + ggtitle("Class, top 100") +
  theme_pubclean() + theme(legend.position="right", axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```


## 6.1 Barplot of relative abundance of *P. protegens*  CHA0


```{r}
#Filter norm tax table for ASV1148 == species "protegens"
ps2.propCHA0 = subset_taxa(ps2.css.prop, Species=="protegens")

plot_bar(ps2.propCHA0, x="id_samples", fill="Species")+
  geom_bar(aes(fill=Species), stat="identity", position="stack")+
  scale_fill_manual(values="#00cc74")+
  xlab("") + ylab("Relative abundance (%)") +
  facet_wrap(~ComState+Environment+Inoculant, scales="free_x", ncol = 4) + ggtitle("Class, top 100") +
  theme_pubclean() + 
  theme(legend.position="right", 
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) 

```

## 6.2 Differences in CSS-normalized relative abundance of CHA0 ASV

```{r}
#Subset for CHA0 with the CSS
ps2.propCHA0 = subset_taxa(ps2.css.prop, Species=="protegens")
CHA0_abund = left_join(data.frame(sample_data(ps2.propCHA0)), rownames_to_column(data.frame(t(otu_table(ps2.propCHA0))), "id_samples"), by=c("id_samples"))

CHA0_abund$StateEnviron = paste0(CHA0_abund$ComState, "_", CHA0_abund$Environment, "_", CHA0_abund$Inoculant)

#stats
kw = agricolae::kruskal(CHA0_abund$ASV1148, CHA0_abund$StateEnviron, p.adj = "BH")
kwres = rownames_to_column(kw$groups, var="StateEnviron")


#Function to calculate number of observations
stat_box_data <- function(y, upper_limit = max(log(CHA0_abund$ASV1148)) -max(log(CHA0_abund$ASV1148))*1.75) {
  return( 
    data.frame(
      y = 0.95 * upper_limit,
      label = paste('n =', length(y), '\n')
    )
  )
}


ggplot(CHA0_abund, aes(x=StateEnviron, y=as.numeric(ASV1148))) +
  geom_boxplot() +
  geom_jitter(color="#00cc74", size=2, shape=16)+
  xlab("") + ylab("Normalized P. protegens ASV relative abundance (%)")+
  annotate(geom = "text", x=kwres$StateEnviron, y=45, label=kwres$groups)+
    ggtitle("Class abundance", subtitle = paste0("Kruskal-Wallis p-value = ",  kw$statistics$p.chisq))+
  scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x), 
                labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") +
  stat_summary(
    fun.data = stat_box_data, 
    geom = "text", 
    hjust = 0.5,
    vjust = 0.9
  ) + 
  theme_pubr() + theme(axis.text.x = element_text(angle = 35, vjust =1, hjust=1))
```


## 6.3 Differences at the Class level.

```{r warning=FALSE}
#merge taxa by Class!
Class_glom = tax_glom(ps2.css.prop, taxrank = "Class")

Class_abund = left_join(data.frame(sample_data(Class_glom)), rownames_to_column(data.frame(t(otu_table(Class_glom))), "id_samples"), by=c("id_samples"))
Class_abund$StateEnviron = paste0(Class_abund$ComState, "_", Class_abund$Environment, "_", Class_abund$Inoculant)
Class_abund$Time = as.character(Class_abund$Time)
Class_abund.melt = Class_abund %>% select(-Replicate, -ng.uL) %>% reshape::melt(value.name = "value")
Class_tax = as.data.frame(tax_table(Class_glom)) %>% rownames_to_column(var = "asvs")
Class_abundtax = left_join(Class_abund.melt, Class_tax, by=c("variable"="asvs"))
head(Class_abundtax)

#Plot all, stored in the classp list of plots
classp = list()

for (i in unique(Class_abundtax_top$Class)){
  class_filt = filter(Class_abundtax_top, Class==i)
  kw = agricolae::kruskal(class_filt$value, class_filt$StateEnviron, p.adj = "BH")
  kwres = rownames_to_column(kw$groups, var="StateEnviron")
  classp[[i]] = ggplot(class_filt, aes(x=StateEnviron, y=value))+
    geom_boxplot(width=0.9)+
    geom_jitter(aes(color=Class), size=0.75, shape=16)+
    annotate(geom = "text", x=kwres$StateEnviron, y= max(class_filt$value)*1.2, label=kwres$groups)+
    xlab("") + ylab("ASVs normalized relative abundance (%)")+
    ggtitle("Class abundance", subtitle = paste0("Kruskal-Wallis p-value = ",  kw$statistics$p.chisq))+
    scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x))+
    annotation_logticks(sides = "l") +
    facet_wrap(~Class, nrow = 1)+
    theme_pubclean() + theme(axis.text.x = element_text(angle = 90, vjust =1, hjust=1))
  
}

classp$Alphaproteobacteria
classp$Bacilli
classp$Gammaproteobacteria
```

## 6.4 Differences in Gammaproteobacteria removing CHA0 counts

P. protegens CHA0 would indeed inflate the gammaproteobacteria relative normalized abundance. If we remove the inoculant counts, would we observe differences in gammaproteobacteria?

```{r}
#Remove CHA0 after normalization
ps2.css_noCHA0 = subset_taxa(ps2.css, Species!="protegens")
ps2.css_noCHA0.prop = transform_sample_counts(ps2.css_noCHA0, function(x) {x/sum(x)*100})

#Frist, merge taxa by Class!
noCHA0_Class_glom = tax_glom(ps2.css_noCHA0.prop, taxrank = "Class")
noCHA0_Class_abund = left_join(data.frame(sample_data(noCHA0_Class_glom)), 
                               rownames_to_column(data.frame(t(otu_table(noCHA0_Class_glom))), "id_samples"), by=c("id_samples"))
noCHA0_Class_abund$StateEnviron = paste0(noCHA0_Class_abund$ComState, "_", 
                                         noCHA0_Class_abund$Environment, "_", 
                                         noCHA0_Class_abund$Inoculant)
noCHA0_Class_abund$Time = as.character(noCHA0_Class_abund$Time)
noCHA0_Class_abund.melt = noCHA0_Class_abund %>% select(-Replicate, -ng.uL) %>% reshape2::melt(value.name = "abund")
noCHA0_Class_tax = as.data.frame(tax_table(noCHA0_Class_glom)) %>% rownames_to_column(var = "asvs")
noCHA0_Class_abundtax = left_join(noCHA0_Class_abund.melt, noCHA0_Class_tax, by=c("variable"="asvs"))

noCHA0_Class_abundtax_top = filter(noCHA0_Class_abundtax, Class=="Gammaproteobacteria")

#Store results

for (i in unique(noCHA0_Class_abundtax_top$Class)){
  class_filt = filter(noCHA0_Class_abundtax_top, Class==i)
  kw = agricolae::kruskal(class_filt$abund, class_filt$StateEnviron, p.adj = "BH")
  kwres = rownames_to_column(kw$groups, var="StateEnviron")
  classp = ggplot(class_filt, aes(x=StateEnviron, y=abund))+
    geom_boxplot(width=0.9)+
    geom_jitter(aes(color=Class), size=0.5, shape=16)+
    annotate(geom = "text", x=kwres$StateEnviron, y=30, label=kwres$groups)+
    xlab("") + ylab("ASVs normalized relative abundance (%)")+
    ggtitle("Class abundance", subtitle = paste0("Kruskal-Wallis p-value = ",  kw$statistics$p.chisq))+
    scale_y_log10(breaks = trans_breaks("log10", function(x) 10^x))+
    annotation_logticks(sides = "l") +
    facet_wrap(~Class, nrow = 1)+
    theme_pubclean() + theme(axis.text.x = element_text(angle = 90, vjust =1, hjust=1))
  
  plot(classp)
  
}


```


# 7. Diversity

## 7.1 Diversity indexes

Observed ASVs, Shannon index and Faith's phylogenetic diversity.

```{r }
#Order by ComState+Environment+Inoculant
order.samples = c("Growing_1_Bulk_CHA0","Growing_5_Bulk_CHA0","Growing_7_Bulk_CHA0","Growing_9_Bulk_CHA0", NULL,
                  "Growing_1_Bulk_noCHA0","Growing_5_Bulk_noCHA0","Growing_7_Bulk_noCHA0","Growing_9_Bulk_noCHA0",
                  "Growing_5_Rhiz_CHA0","Growing_7_Rhiz_CHA0","Growing_9_Rhiz_CHA0",
                  "Growing_5_Rhiz_noCHA0","Growing_7_Rhiz_noCHA0","Growing_9_Rhiz_noCHA0",
                  "Stable_1_Bulk_CHA0","Stable_5_Bulk_CHA0","Stable_7_Bulk_CHA0","Stable_9_Bulk_CHA0",
                  "Stable_1_Bulk_noCHA0","Stable_5_Bulk_noCHA0","Stable_7_Bulk_noCHA0","Stable_9_Bulk_noCHA0",
                  "Stable_5_Rhiz_CHA0","Stable_7_Rhiz_CHA0","Stable_9_Rhiz_CHA0",
                  "Stable_5_Rhiz_noCHA0","Stable_7_Rhiz_noCHA0","Stable_9_Rhiz_noCHA0")

#let's also add the corresponding colors.
order.cols = data.frame("Names"=order.samples) %>% left_join(metadata, by="Names") %>% 
  dplyr::select(-id_samples, -ComState, -Inoculant, -Environment, -Time, -Replicate, -Names_rep) %>% unique()

#Observed ASVs
diver.ob = plot_richness(ps2, x="Names", measures=c("Observed"))+
  geom_bar(position = "identity", stat = "summary", aes(y=value-100), fill="darkgrey")+
  geom_errorbar(stat = "summary", width=0.75, aes(y=value-100)) + 
  geom_jitter(width = 0.2, alpha=0.75, size=1, aes(y=value-100))+
  scale_y_continuous(breaks=c(0,100,200), labels=c(100, 200, 300))+
  facet_wrap(~variable,scales = "free_x", nrow = 1)+
  ggpubr::rotate() +
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1))

diver.ob$layers = diver.ob$layers[-1]
diver.ob$data$Names <- as.character(diver.ob$data$Names)
diver.ob$data$Names <- factor(diver.ob$data$Names, levels=rev(order.samples))

#calculate shannon independently for better representation
diver.sha = plot_richness(ps2, x="Names", measures=c("Shannon"))+
  geom_bar(position = "identity", stat = "summary", aes(y=value-2), fill="darkgrey")+
  geom_errorbar(stat = "summary", width=0.75, aes(y=value-2)) + 
  geom_jitter(width = 0.2, alpha=0.75, size=1, aes(y=value-2))+
  scale_y_continuous(breaks=c(0.0,0.5,1.0,1.5,2.0), labels=c(2, 2.5, 3,3.5,4))+
  facet_wrap(~variable,scales = "free_x", nrow = 1)+
  ggpubr::rotate() +
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1))
diver.sha$layers = diver.sha$layers[-1]
diver.sha$data$Names <- as.character(diver.sha$data$Names)
diver.sha$data$Names <- factor(diver.sha$data$Names, levels=rev(order.samples))

#and Faith's PD
pd = pd(otu_table(ps2), phy_tree(ps2), include.root = F)
pd = pd %>% rownames_to_column("id_samples") %>% left_join(metadata, by="id_samples")
pd$Names = gsub("_[123456]$", "", pd$Names)
diver.pd = ggplot(pd, aes(Names, PD)) + 
  geom_bar(position = "identity", stat = "summary", aes(y=PD-20), fill="darkgrey")+
  geom_errorbar(stat = "summary",width=0.75, aes(y=PD-20)) + 
  geom_jitter(width = 0.2, alpha=0.75, size=1, aes(y=PD-20))+
  scale_y_continuous(breaks=c(0,10,20), labels=c(20, 30, 40))+
  facet_wrap(NULL, labeller=as_labeller(c("(all)"="Faith's PD")))+
  ggpubr::rotate() +
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1)) + xlab("")
  
diver.pd$data$Names = as.character(diver.pd$data$Names)
diver.pd$data$Names = factor(diver.pd$data$Names, levels=rev(order.samples))

ggarrange(diver.ob+theme(legend.position = "", axis.line = element_line())+scale_color_manual(values=rev(order.cols$Cols)),
          diver.sha+theme(legend.position = "", axis.line = element_line())+scale_color_manual(values=rev(order.cols$Cols)),
          diver.pd+theme(legend.position = "", axis.line = element_line())+scale_color_manual(values=rev(order.cols$Cols)), 
          ncol=3, widths = c(1,1,1), common.legend = T, legend = "bottom", align = "hv")


#Save all diversity values in a table for manual inspection
diversity_all = pd %>% dplyr::select(id_samples, Names, PD)  %>% 
  left_join(select(diver.ob$data, id_samples, Observed_ASVs = value), by = c("id_samples")) %>%
  left_join(select(diver.sha$data, id_samples, Shannon = value), by = c("id_samples"))

#write.table(diversity_all, "Diversity_indexes.txt", quote = F, sep = "\t", row.names = F)

```

## 7.2 Differences in diversity across comm. state, environments or inoculation

```{r }
#Observed ASVs
diver.ob.comstate = plot_richness(ps2, x="ComState", measures=c("Observed"))+geom_boxplot()+
  ylab("Observed ASVs")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("Growing", "Stable")), label = "p.format") +
  facet_wrap(~variable,scales = "free_x", nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.ob.comstate$data$Names <- as.character(diver.ob.comstate$data$Names)
diver.ob.comstate$data$Names <- factor(diver.ob.comstate$data$Names, levels=rev(order.samples))

diver.ob.environ = plot_richness(ps2, x="Environment", measures=c("Observed"))+geom_boxplot()+
  ylab("Observed ASVs")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("Bulk", "Rhiz")), label = "p.format") +
  facet_wrap(~variable+ComState,scales = "free_x", nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.ob.environ$data$Names <- as.character(diver.ob.environ$data$Names)
diver.ob.environ$data$Names <- factor(diver.ob.environ$data$Names, levels=rev(order.samples))

diver.ob.inoculant = plot_richness(ps2, x="Inoculant", measures=c("Observed"))+geom_boxplot()+
  ylab("Observed ASVs")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("CHA0", "noCHA0")), label = "p.format") +
  facet_wrap(~variable+ComState+Environment,scales = "free_x", nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.ob.inoculant$data$Names <- as.character(diver.ob.inoculant$data$Names)
diver.ob.inoculant$data$Names <- factor(diver.ob.inoculant$data$Names, levels=rev(order.samples))

observed_all = ggarrange(diver.ob.comstate, 
          diver.ob.environ+ylab("")+theme(axis.text.y=element_blank()), 
          diver.ob.inoculant+ylab("")+theme(axis.text.y=element_blank()),
          nrow = 1, align = "h",
          widths = c(4,5.5,9.5))

#SHANNON ASVs
diver.sha.comstate = plot_richness(ps2, x="ComState", measures=c("Shannon"))+geom_boxplot()+
  ylab("SHannon")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("Growing", "Stable")), label = "p.signif") +
  facet_wrap(~variable,scales = "free_x", nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.sha.comstate$data$Names <- as.character(diver.sha.comstate$data$Names)
diver.sha.comstate$data$Names <- factor(diver.sha.comstate$data$Names, levels=rev(order.samples))

diver.sha.environ = plot_richness(ps2, x="Environment", measures=c("Shannon"))+geom_boxplot()+
  ylab("Shannon")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("Bulk", "Rhiz")), label = "p.signif") +
  facet_wrap(~variable+ComState,scales = "free_x", nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.sha.environ$data$Names <- as.character(diver.sha.environ$data$Names)
diver.sha.environ$data$Names <- factor(diver.sha.environ$data$Names, levels=rev(order.samples))

diver.sha.inoculant = plot_richness(ps2, x="Inoculant", measures=c("Shannon"))+geom_boxplot()+
  ylab("Shannon")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("CHA0", "noCHA0")), label = "p.signif") +
  facet_wrap(~variable+ComState+Environment,scales = "free_x", nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.sha.inoculant$data$Names <- as.character(diver.sha.inoculant$data$Names)
diver.sha.inoculant$data$Names <- factor(diver.sha.inoculant$data$Names, levels=rev(order.samples))

shannon_all = ggarrange(diver.sha.comstate, 
          diver.sha.environ+ylab("")+theme(axis.text.y=element_blank()), 
          diver.sha.inoculant+ylab("")+theme(axis.text.y=element_blank()),
          nrow = 1, align = "h",
          widths = c(4,5.5,9.5))

#PD
diver.pd.comstate = ggplot(pd, aes(ComState, PD)) + geom_boxplot() +
  ylab("Faith's PD")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("Growing", "Stable")), label = "p.format") +
  facet_wrap(NULL, labeller=as_labeller(c("(all)"="Faith's PD")), nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.pd.comstate$data$Names = as.character(diver.pd.comstate$data$Names)
diver.pd.comstate$data$Names = factor(diver.pd.comstate$data$Names, levels=rev(order.samples))

diver.pd.environ = ggplot(pd, aes(Environment, PD)) + geom_boxplot() +
  ylab("Faith's PD")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("Bulk", "Rhiz")), label = "p.format") +
  facet_wrap(~ComState, nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.pd.environ$data$Names = as.character(diver.pd.environ$data$Names)
diver.pd.environ$data$Names = factor(diver.pd.environ$data$Names, levels=rev(order.samples))

diver.pd.inoculant = ggplot(pd, aes(Inoculant, PD)) + geom_boxplot() +
  ylab("Faith's PD")+
  stat_compare_means(method="wilcox.test", comparisons = list(c("CHA0", "noCHA0")), label = "p.format") +
  facet_wrap(~ComState+Environment, nrow = 1)+
  theme_pubclean() + theme(legend.position = "bottom", 
                           axis.text.x = element_text(angle = 90, vjust = 0, hjust=1),
                           axis.line = element_line())

diver.pd.inoculant$data$Names = as.character(diver.pd.inoculant$data$Names)
diver.pd.inoculant$data$Names = factor(diver.pd.inoculant$data$Names, levels=rev(order.samples))

pd_all = ggarrange(diver.pd.comstate, 
          diver.pd.environ+ylab("")+theme(axis.text.y=element_blank()), 
          diver.pd.inoculant+ylab("")+theme(axis.text.y=element_blank()),
          nrow = 1, align = "h",
          widths = c(4,5.5,9.5))

ggarrange(observed_all, shannon_all, pd_all, nrow = 3)
```

## 7.3 Correlation of diversity vs. time


```{r }
library(ggpmisc)

#SHANNON
ggplot(diver.sha$data, aes(x=as.numeric(Time), y=value, color=ComState, fill=ComState)) + geom_point() + ylab("Shannon") +
  xlab("Time (days)") + ggtitle("Spearman correlation")+
  geom_smooth(method = "gam", formula = y ~ s(x, bs = "cs", k=3), size=0.5) +
  stat_cor(method = "spearman") +
  scale_color_manual(values=c("#28511b", "#ccae00"))+
  scale_x_continuous(breaks = c(1,2,3,4,5,6,7,8,9), labels = c(1,"","","",5,"",7,"",9))+
  scale_fill_manual(values=c("#28511b", "#ccae00"))+
  facet_wrap(~Inoculant+Environment, scales = "free_x")+
  theme_pubclean()+ theme(axis.line = element_line())

#PD
ggplot(diver.pd$data, aes(x=as.numeric(Time), y=PD, color=ComState, fill=ComState)) + geom_point() + ylab("Faith's PD") +
  xlab("Time (days)") + ggtitle("Spearman correlation")+
  geom_smooth(method = "gam", formula = y ~ s(x, bs = "cs", k=3), size=0.5) +
  stat_cor(method = "spearman") +
  scale_color_manual(values=c("#28511b", "#ccae00"))+
  scale_x_continuous(breaks = c(1,2,3,4,5,6,7,8,9), labels = c(1,"","","",5,"",7,"",9))+
  scale_fill_manual(values=c("#28511b", "#ccae00"))+
  facet_wrap(~Inoculant+Environment, scales = "free_x")+
  theme_pubclean()+ theme(axis.line = element_line())

#Observed ASVs
ggplot(diver.ob$data, aes(x=as.numeric(Time), y=value, color=ComState, fill=ComState)) + geom_point() + ylab("Observed ASVs") +
  xlab("Time (days)") + ggtitle("Spearman correlation")+
  geom_smooth(method = "gam", formula = y ~ s(x, bs = "cs", k=3), size=0.5) +
  stat_cor(method = "spearman") +
  scale_color_manual(values=c("#28511b", "#ccae00"))+
  scale_x_continuous(breaks = c(1,2,3,4,5,6,7,8,9), labels = c(1,"","","",5,"",7,"",9))+
  scale_fill_manual(values=c("#28511b", "#ccae00"))+
  facet_wrap(~Inoculant+Environment, scales = "free_x")+
  theme_pubclean()+ theme(axis.line = element_line())

```


# 8. NMDS


```{r}
set.seed(1234)
#Individually NMDS by ComState
#Growing
ps2.prop.grw = subset_samples(ps2.prop, ComState =="Growing")
ord.nmds.bray.grw = ordinate(ps2.prop.grw, method="NMDS", distance="bray", k=2, trymax=500)
stressplot(ord.nmds.bray.grw)
scrs.bray.grw = scores(ord.nmds.bray.grw, display = 'sites')
scrs.bray.grw <- cbind(as.data.frame(scrs.bray.grw), 
                       Names_rep = ps2.prop.grw@sam_data$Names_rep, 
                       Names = ps2.prop.grw@sam_data$Names, 
                       Inoculant = ps2.prop.grw@sam_data$Inoculant, 
                       ComState = ps2.prop.grw@sam_data$ComState, 
                       Env =ps2.prop.grw@sam_data$Environment)
cen.bray.grw = aggregate(cbind(NMDS1, NMDS2) ~ Names, data = scrs.bray.grw, FUN = mean)
segs.bray.grw = merge(scrs.bray.grw, setNames(cen.bray.grw, c('Names', 'oNMDS1','oNMDS2')),
              by = 'Names', sort = FALSE)

grw = plot_ordination(ps2.prop.grw, ord.nmds.bray.grw, color="Names", title="Bray NMDS | Growing com.") +
  geom_point(alpha=0.1) +
  geom_segment(data = segs.bray.grw, mapping = aes(xend = oNMDS1, yend = oNMDS2), size=0.6, alpha=0.25) +
  geom_point(data = cen.bray.grw, size = 3) +
  geom_hline(yintercept = 0, linetype="dashed", color="grey") +
  geom_vline(xintercept = 0, linetype="dashed", color="grey") +
  scale_color_manual(values=c("#dbd8bf", "#d8bfdc", "#f5edc7", "#f498c1","#c8e8e4", "#d9daea", "#e0d085", "#ea519d","#80cdc1", "#b2abd2", "#be9736", "#d42e92","#35978f", "#8074ad"))+
  annotate("text", size=3, y=-0.07, x=-0.25,
           label = paste0("Stress: ", format(round(ord.nmds.bray.grw$stress, 5))), hjust=0)+
  theme_classic2() + theme(legend.position="bottom")

#Stable
ps2.prop.stb = subset_samples(ps2.prop, ComState =="Stable")
ord.nmds.bray.stb = ordinate(ps2.prop.stb, method="NMDS", distance="bray", k=2, trymax=500)
stressplot(ord.nmds.bray.stb)
scrs.bray.stb = scores(ord.nmds.bray.stb, display = 'sites')
scrs.bray.stb <- cbind(as.data.frame(scrs.bray.stb), 
                       Names_rep = ps2.prop.stb@sam_data$Names_rep, 
                       Names = ps2.prop.stb@sam_data$Names, 
                       Inoculant = ps2.prop.stb@sam_data$Inoculant, 
                       ComState = ps2.prop.stb@sam_data$ComState, 
                       Env =ps2.prop.stb@sam_data$Environment)
cen.bray.stb = aggregate(cbind(NMDS1, NMDS2) ~ Names, data = scrs.bray.stb, FUN = mean)
segs.bray.stb = merge(scrs.bray.stb, setNames(cen.bray.stb, c('Names', 'oNMDS1','oNMDS2')),
              by = 'Names', sort = FALSE)

stb = plot_ordination(ps2.prop.stb, ord.nmds.bray.stb, color="Names", title="Bray NMDS | Stable com.") +
  geom_point(alpha=0.1) +
  geom_segment(data = segs.bray.stb, mapping = aes(xend = oNMDS1, yend = oNMDS2), size=0.6, alpha=0.25) +
  geom_point(data = cen.bray.stb, size = 3) +
  geom_hline(yintercept = 0, linetype="dashed", color="grey") +
  geom_vline(xintercept = 0, linetype="dashed", color="grey") +
  scale_color_manual(values=c("#c39fcb", "#ddbfb8", "#fce0ec", "#fddbc7","#e6f1d0", "#d1e5ef", "#edb5d3", "#f4a582","#b9d987", "#93c5de", "#dd77ae", "#d6604d","#80bc42", "#4393c3"))+
  annotate("text", size=3, y=-0.07, x=-0.25,
           label = paste0("Stress: ", format(round(ord.nmds.bray.stb$stress, 5))), hjust=0)+
  theme_classic2() + theme(legend.position="bottom")

grw
stb

```

Calculate density graphs for the Stable/Growing conditions to better see differences:

```{r}
#Growing density plots per NMDS
grw.den.1 = ggplot(scrs.bray.grw, aes(x=NMDS1,color=Names, fill=Names)) +
  geom_density(alpha=0.5) +
  scale_fill_manual(values=c("#dbd8bf", "#d8bfdc", "#f5edc7", "#f498c1",
                             "#c8e8e4", "#d9daea", "#e0d085", "#ea519d",
                             "#80cdc1", "#b2abd2", "#be9736", "#d42e92",
                             "#35978f", "#8074ad"))+
  scale_color_manual(values=c("#dbd8bf", "#d8bfdc", "#f5edc7", "#f498c1",
                              "#c8e8e4", "#d9daea", "#e0d085", "#ea519d", 
                              "#80cdc1", "#b2abd2", "#be9736", "#d42e92",
                              "#35978f", "#8074ad"))+
  theme_pubclean() + theme(legend.position="none")

grw.den.2 = ggplot(scrs.bray.grw, aes(y=NMDS2,color=Names, fill=Names)) +
  geom_density(alpha=0.5) +
  scale_fill_manual(values=c("#dbd8bf", "#d8bfdc", "#f5edc7", "#f498c1",
                             "#c8e8e4", "#d9daea", "#e0d085", "#ea519d",
                             "#80cdc1", "#b2abd2", "#be9736", "#d42e92",
                             "#35978f", "#8074ad"))+
  scale_color_manual(values=c("#dbd8bf", "#d8bfdc", "#f5edc7", "#f498c1",
                              "#c8e8e4", "#d9daea", "#e0d085", "#ea519d",
                              "#80cdc1", "#b2abd2", "#be9736", "#d42e92",
                              "#35978f", "#8074ad"))+
  theme_pubclean()+ theme(legend.position="none")


#Stable density plots per NMDS
stb.den.1 = ggplot(scrs.bray.stb, aes(x=NMDS1,color=Names, fill=Names)) +
  geom_density(alpha=0.5) +
  scale_fill_manual(values=c("#c39fcb", "#ddbfb8", "#fce0ec", "#fddbc7",
                             "#e6f1d0", "#d1e5ef", "#edb5d3", "#f4a582",
                             "#b9d987", "#93c5de", "#dd77ae", "#d6604d",
                             "#80bc42", "#4393c3"))+
  scale_color_manual(values=c("#c39fcb", "#ddbfb8", "#fce0ec", "#fddbc7",
                              "#e6f1d0", "#d1e5ef", "#edb5d3", "#f4a582",
                              "#b9d987", "#93c5de", "#dd77ae", "#d6604d",
                              "#80bc42", "#4393c3"))+
  theme_pubclean()+ theme(legend.position="none")

stb.den.2 = ggplot(scrs.bray.stb, aes(y=NMDS2,color=Names, fill=Names)) +
  geom_density(alpha=0.5) +
  scale_fill_manual(values=c("#c39fcb", "#ddbfb8", "#fce0ec", "#fddbc7",
                             "#e6f1d0", "#d1e5ef", "#edb5d3", "#f4a582",
                             "#b9d987", "#93c5de", "#dd77ae", "#d6604d",
                             "#80bc42", "#4393c3"))+
  scale_color_manual(values=c("#c39fcb", "#ddbfb8", "#fce0ec", "#fddbc7",
                              "#e6f1d0", "#d1e5ef", "#edb5d3", "#f4a582",
                              "#b9d987", "#93c5de", "#dd77ae", "#d6604d",
                              "#80bc42", "#4393c3"))+
  theme_pubclean()+ theme(legend.position="none")


#And plot both
ggarrange(ggarrange(stb.den.1+xlab("") +theme(axis.text.x=element_blank()) + theme_classic2(), 
                              NA, 
                              stb + theme(legend.position="none") + ggtitle("Stable"), 
                              stb.den.2+ylab("") +theme(axis.text.y=element_blank())+ theme_classic2(), 
                              ncol=2, nrow = 2, 
                              heights = c(0.85, 2),
                              widths  = c(2, 0.85),
                              align = "hv", 
                              common.legend = TRUE, legend = "right"))


ggarrange(ggarrange(grw.den.1+xlab("") +theme(axis.text.x=element_blank()) + theme_classic2(), 
                              NA, 
                              grw + theme(legend.position="none") + ggtitle("Growing"), 
                              grw.den.2+ylab("") +theme(axis.text.y=element_blank())+ theme_classic2(), 
                              ncol=2, nrow = 2, 
                              heights = c(0.85, 2),
                              widths  = c(2, 0.85),
                              align = "hv", 
                              common.legend = TRUE, legend = "right"))

```




## 8.1 Distance to the centroid

```{r}
#Growing NatCom
cenbeta.grw = vegan::betadisper(phyloseq::distance(ps2.prop.grw, "bray"), 
                            as.factor(scrs.bray.grw$Names), 
                            type = "centroid", sqrt.dist = F, bias.adjust = T)
cenbeta.grw = as.data.frame(cenbeta.grw$distances)
colnames(cenbeta.grw) = "distances"
cenbeta.grw = rownames_to_column(cenbeta.grw, "id_samples") %>% left_join(metadata, by="id_samples")

cen.dist.grw.plot = ggplot(cenbeta.grw, aes(x=Names, y=distances)) + 
  geom_bar(position = "identity", stat = "summary", fill="darkgrey") + 
  ggtitle("Growng NatCom")+
  geom_errorbar(stat="summary", width=0.5)+
  geom_jitter(shape=16, width = 0.2) +
  theme_pubclean() + ylab("Distance to centroid") +
  facet_wrap(~Environment+Inoculant, scales = "free_x", ncol = 4)+
  theme(axis.line = element_line(), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

#Stable NatCom
cenbeta.stb = vegan::betadisper(phyloseq::distance(ps2.prop.stb, "bray"), 
                            as.factor(scrs.bray.stb$Names), 
                            type = "centroid", sqrt.dist = F, bias.adjust = T)
cenbeta.stb = as.data.frame(cenbeta.stb$distances)
colnames(cenbeta.stb) = "distances"
cenbeta.stb = rownames_to_column(cenbeta.stb, "id_samples") %>% left_join(metadata, by="id_samples")

cen.dist.stb.plot = ggplot(cenbeta.stb, aes(x=Names, y=distances)) + 
  geom_bar(position = "identity", stat = "summary", fill="darkgrey") + 
  ggtitle("Stable NatCom")+
  geom_errorbar(stat="summary", width=0.5)+
  geom_jitter(shape=16, width = 0.2) +
  theme_pubclean() + ylab("Distance to centroid") +
  facet_wrap(~Environment+Inoculant, scales = "free_x", ncol = 4)+
  theme(axis.line = element_line(), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

ggarrange(cen.dist.grw.plot, cen.dist.stb.plot, nrow = 1)
```

## 8.2 Bray-Curtis statistical differences

Test for homogeneity of the data

```{r}
bray.rep=phyloseq::distance(ps2.prop, method="bray")
sampledf = data.frame(sample_data(ps2.prop))

set.seed(1234)
permutest(betadisper(bray.rep, sampledf$Names))
```

Our data is in the limit of being homogeneously dispersed. ANOVA is generally robust to violations of the assumption (homogeneity) as long as group sizes are equal. Group sizes are indeed equal. Therefore we will use PERMANOVA (permutational multivariate ANOVA (analysis of variance) and also kruskal in case there are discrepancies.

### 8.2.1 Growing vs. Stable communities?

```{r}
#Are grow/Stable are different?
adonis2(formula = bray.rep ~ ps2.prop@sam_data$ComState, data = sampledf, permutations = 9999)
kruskal.test(ps2.prop@sam_data$ComState ~ ps2.prop@sam_data$Names, data=as.matrix(bray.rep))
```

Yes, there are.

### 8.2.2 Bulk vs. Rhizosphere


```{r}
#Obtain bray-curtis distances of the growing community
bray.grw=phyloseq::distance(ps2.prop.grw, method="bray")
sampledf.grw = data.frame(sample_data(ps2.prop.grw))

#Obtain bray-curtis distances of the Stable community
bray.stb=phyloseq::distance(ps2.prop.stb, method="bray")
sampledf.stb = data.frame(sample_data(ps2.prop.stb))

#Are Bulk / Rhizosphere different in growing?
adonis2(formula = bray.grw ~ ps2.prop.grw@sam_data$Environment, data = sampledf.grw, permutations = 9999)
kruskal.test(ps2.prop.grw@sam_data$Environment ~ ps2.prop.grw@sam_data$Names, data=as.matrix(bray.grw))
#Are inoculated no inoculated different in Stable?
adonis2(formula = bray.stb ~ ps2.prop.stb@sam_data$Environment, data = sampledf.stb, permutations = 9999)
kruskal.test(ps2.prop.stb@sam_data$Environment ~ ps2.prop.stb@sam_data$Names, data=as.matrix(bray.stb))
```

Yes. In both cases, there are differences between bulk and rhizosphere conditions, in the growing or stable NatCom.

### 8.2.3 Inocualted vs. not inoculate in each ComState and Environment

Are there differences between inoculated or non-inoculated within each community state (growing and Stable) and depending if we consider the Rhizosphere or the Bulk?

```{r}
#Are inoculated no inoculated different in GROWING?
adonis2(formula = bray.grw ~ ps2.prop.grw@sam_data$Inoculant * ps2.prop.grw@sam_data$Environment, data = sampledf.grw, permutations = 9999)

#Are inoculated no inoculated different in Stable?
adonis2(formula = bray.stb ~ ps2.prop.stb@sam_data$Inoculant * ps2.prop.stb@sam_data$Environment, data = sampledf.stb, permutations = 9999)
```

## 8.3 Correlation of Bray-Curtis vs. time

```{r )}
library(ggpmisc)
ggplot(filter(bray.rep.info, ComState.x == ComState.y & Environment.x == Environment.y & Inoculant.x == Inoculant.y & Time.x == Time.y),
              aes(x=as.numeric(Time.x), y=BC_diss, color=ComState.x, fill=ComState.x)) + 
  geom_point() + ylab("Bray-Curtis dissimilarity") +
  xlab("Time (days)") + ggtitle("Spearman correlation")+
  geom_smooth(method = "gam", formula = y ~ s(x, bs = "cs", k=3), size=0.5) +
  stat_cor(method = "spearman") +
  scale_color_manual(values=c("#8075ae", "#4393c4"))+
  scale_x_continuous(breaks = c(1,2,3,4,5,6,7,8,9), labels = c(1,"","","",5,"",7,"",9))+
  scale_fill_manual(values=c("#8075ae", "#4393c4"))+
  facet_wrap(~Inoculant.x+Environment.x, scales = "free_x")+
  theme_pubclean()+ theme(axis.line = element_line())
```



# 9. *Pseudomonas*

## 9.1 Relative abundance of top 5 *Pseudomonas* ASVs

```{r}
ps2.prop = transform_sample_counts(ps2, function(x) {x/sum(x)*100})
#Filter tax table for Pseudomonas ASVs
ps2.propPseudo = subset_taxa(ps2.prop, Genus=="Pseudomonas")
#Write all pseudomonas refseqs:
write.table(refseq(ps2.propPseudo), "ref-seq_PSEUDO_all.txt", sep = "\t", quote = F, col.names = F)

#retrieve top 5
toptax.Pseudo = names(sort(taxa_sums(ps2.propPseudo), decreasing=TRUE))[1:5]
ps.toptax.Pseudo = prune_taxa(toptax.Pseudo, ps2.propPseudo)

Pseudo = as.data.frame(otu_table(ps.toptax.Pseudo))
Pseudo = rownames_to_column(Pseudo, "id_samples")
Pseudo.melt = reshape2::melt(Pseudo)
Pseudo.melt = Pseudo.melt %>% left_join(data.frame(sample_data(ps2.propPseudo)), by="id_samples")

ggplot(filter(Pseudo.melt, Time %in% c(1,9)), aes(x=Names, y=value/4))+
  geom_bar(aes(fill=variable), stat="identity")+ ylab("Normalized relative abundance")+
  scale_fill_manual(values=c("#e0bd89", "#31a872", "#3b79bc", "purple", "#c42c2c"))+
  facet_grid2(~ComState+Inoculant+Environment, scales="free_x", space = T)+
  theme_pubclean() + theme(legend.position="right", axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) 

```

## 9.2 Differences in relative abundance of *Pseudomonas*

```{r}
ps2.css.propPseudo = subset_taxa(ps2.css.prop, Genus=="Pseudomonas")
#Write all pseudomonas refseqs:
#write.table(refseq(ps2.css.propPseudo), "ref-seq_PSEUDO_all.txt", sep = "\t", quote = F, col.names = F)

#retrieve top 5
toptax.css.Pseudo = names(sort(taxa_sums(ps2.css.propPseudo), decreasing=TRUE))[1:5]
ps.css.toptax.Pseudo = prune_taxa(toptax.css.Pseudo, ps2.css.propPseudo)

Pseudo.css = as.data.frame(otu_table(ps.css.toptax.Pseudo))
Pseudo.css = rownames_to_column(Pseudo.css, "id_samples")
Pseudo.css.melt = reshape2::melt(Pseudo.css)
Pseudo.css.melt = Pseudo.css.melt %>% 
  left_join(data.frame(sample_data(ps2.css.propPseudo)), by= join_by("variable" == "id_samples"))
head(Pseudo.css.melt)
```


```{r warning=FALSE}
Pseudo.melt.filtered = filter(Pseudo.css.melt, id_samples %in% c("ASV1142","ASV1148", "ASV1168", "ASV1169", "ASV1173") & Time %in% c(1,9))


plots_ps = list()

for (i in c("ASV1142","ASV1148", "ASV1168", "ASV1169", "ASV1173")){
  pseudo_name = i
  df = filter(Pseudo.melt.filtered, id_samples == i)
  kw = agricolae::kruskal(df$value, df$Names, p.adj = "BH")
  kwres = rownames_to_column(kw$groups, var="StateEnviron")
  
  plots_ps[[i]] = ggplot(df, aes(x=Names, y=value)) +
  #geom_bar(position = "identity", stat = "summary", fill="darkgrey", width = 0.7, fun = mean) +
  geom_segment(aes(y = value, yend = 0), stat = "summary", 
               position = "identity", width = 0.7, fun = mean, size = 15, color="darkgrey")+
  geom_jitter(shape=16, width = 0.2)+
  annotate(geom = "text", x=kwres$StateEnviron, y=100, label=kwres$groups)+
  xlab("Sampling time (days)") + ylab("Normalized relative abundance (%)")+
  ggtitle("CHA0 abundance",  subtitle = paste0("Kruskal-Wallis p-value = ",  kw$statistics$p.chisq))+
  scale_y_log10(limits = c(0.001, 100),
                breaks = trans_breaks("log10", function(x) 10^x), 
                labels = trans_format("log10", math_format(10^.x))) +
  geom_errorbar(stat="summary", width=0.3) +
  annotation_logticks(sides = "l") +
  theme_pubr() + 
  theme(axis.text.x = element_blank())
}

plots_ps

```
